### DISCUSSION TITLE
Mikro orm property initalizers not working when using minifed builds

### DISCUSSION THREAD

**Discussion by alanskovrlj** [mikro-orm/mikro-orm#5640] [General]
### Describe the bug

I'm having problems working with entity constructors when using minified builds. When I try to construct a new entity with only the required data, MikroORM seems to be using the entire entity object as the default value for the rest of the fields.

Here is an example:

_src/index.ts_
```typescript
const { MikroORM } = require("@mikro-orm/postgresql"); // or any other driver package
import { User } from './entities/User'

async function run () {
    const orm = await MikroORM.init({
        dbName: "postgres",
        password: "mysecretpassword",
        allowGlobalContext: true,
        entities: [User],
    })

    const user = orm.em.create(User, { name: "John" })
    console.log(user);
    await orm.em.persistAndFlush(user)
}

run()
``` 

_src/entities/User.ts_
```typescript
import {
    Entity,
    PrimaryKey,
    Property,
} from '@mikro-orm/core'

import { randomUUID } from 'crypto'

@Entity()
export class User {
    @PrimaryKey({ defaultRaw: 'uuid_generate_v4()', type: 'uuid' })
    public id: string

    @Property({ type: 'string', unique: true })
    public name: string

    @Property({ type: 'string', nullable: true})
    public age?: string

    constructor(name: string, age: string) {
        this.id = randomUUID()
        this.name = name
        this.age = age
    }
}
``` 

The code above results in following object being created:

```javascript
User {
  name: 'John',
  age: { name: 'John' },
  id: '533de9da-fcfe-4576-8acb-a4c2a60a7bc2'
}
``` 

I believe this is not the intended behavior because when I create a non-minified bundle, it results in:
```javascript
User { name: 'John', id: '86122711-db55-463f-901c-595f097fa081' }
``` 

I am using esbuild to create the bundle, and the only difference between the two examples is the minified flag.

### Reproduction

I've included the reproduction steps above, but I can upload my reproduction code and link if needed.

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.8

### Node.js version

v20.10.0

### Operating system

Linux

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
I don't think you can minify your code and expect this to work, it will result in changed parameter names and those need to be aligned with the property names. If you provide complete reproduction, I can have a look, but I am almost sure it's about this. Take a look at what gets compiled for the entity, if the constructor parameters have wrong names, that's what you get.

One way around this could be using metadata cache which you would generate upfront, so from your entities without minification. 

https://mikro-orm.io/docs/guide/advanced#deployment

**Comment by alanskovrlj** [comment]
Thanks for explaining. For anyone else encountering this issue using esbuild, set minifyIdentifiers: false in your esbuild config. This will still minify syntax and whitespace but will fix the issue mentioned above.
